# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStoreParent.h
# Commit: b668d0c685e8
# Full Hash: b668d0c685e8d0bbcd3bf5b1856389ca41be31f3
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2025-01-09 04:16:57
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 5: Exempt third-party cookie blocking if the channel is on the exception list. r=cookie-reviewers,bvandersloot,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D223384
# ==============================================================================

diff -r 08acd8256071 -r b668d0c685e8 dom/cookiestore/CookieStoreParent.h
--- a/dom/cookiestore/CookieStoreParent.h	Wed Jan 08 19:25:59 2025 +0000
+++ b/dom/cookiestore/CookieStoreParent.h	Wed Jan 08 19:25:59 2025 +0000
@@ -34,24 +34,25 @@
       const nsString& aDomain, const OriginAttributes& aOriginAttributes,
       const Maybe<OriginAttributes>& aPartitionedOriginAttributes,
       const bool& aThirdPartyContext, const bool& aPartitionForeign,
-      const bool& aUsingStorageAccess, const bool& aMatchName,
-      const nsString& aName, const nsCString& aPath,
+      const bool& aUsingStorageAccess, const bool& aIsOn3PCBExceptionList,
+      const bool& aMatchName, const nsString& aName, const nsCString& aPath,
       const bool& aOnlyFirstMatch, GetRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvSetRequest(
       const nsString& aDomain, const OriginAttributes& aOriginAttributes,
       const bool& aThirdPartyContext, const bool& aPartitionForeign,
-      const bool& aUsingStorageAccess, const nsString& aName,
-      const nsString& aValue, const bool& aSession, const int64_t& aExpires,
-      const nsString& aPath, const int32_t& aSameSite, const bool& aPartitioned,
-      const nsID& aOperationID, SetRequestResolver&& aResolver);
+      const bool& aUsingStorageAccess, const bool& aIsOn3PCBExceptionList,
+      const nsString& aName, const nsString& aValue, const bool& aSession,
+      const int64_t& aExpires, const nsString& aPath, const int32_t& aSameSite,
+      const bool& aPartitioned, const nsID& aOperationID,
+      SetRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvDeleteRequest(
       const nsString& aDomain, const OriginAttributes& aOriginAttributes,
       const bool& aThirdPartyContext, const bool& aPartitionForeign,
-      const bool& aUsingStorageAccess, const nsString& aName,
-      const nsString& aPath, const bool& aPartitioned, const nsID& aOperationID,
-      DeleteRequestResolver&& aResolver);
+      const bool& aUsingStorageAccess, const bool& aIsOn3PCBExceptionList,
+      const nsString& aName, const nsString& aPath, const bool& aPartitioned,
+      const nsID& aOperationID, DeleteRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvClose();
 
@@ -59,8 +60,9 @@
       const nsAString& aDomain, const OriginAttributes& aOriginAttributes,
       const Maybe<OriginAttributes>& aPartitionedOriginAttributes,
       bool aThirdPartyContext, bool aPartitionForeign, bool aUsingStorageAccess,
-      bool aMatchName, const nsAString& aName, const nsACString& aPath,
-      bool aOnlyFirstMatch, nsTArray<CookieData>& aResults);
+      bool aIsOn3PCBExceptionList, bool aMatchName, const nsAString& aName,
+      const nsACString& aPath, bool aOnlyFirstMatch,
+      nsTArray<CookieData>& aResults);
 
   // Returns true if a cookie notification has been generated while completing
   // the operation.
@@ -68,18 +70,20 @@
                               const nsAString& aDomain,
                               const OriginAttributes& aOriginAttributes,
                               bool aThirdPartyContext, bool aPartitionForeign,
-                              bool aUsingStorageAccess, const nsAString& aName,
-                              const nsAString& aValue, bool aSession,
-                              int64_t aExpires, const nsAString& aPath,
-                              int32_t aSameSite, bool aPartitioned,
-                              const nsID& aOperationID);
+                              bool aUsingStorageAccess,
+                              bool aIsOn3PCBExceptionList,
+                              const nsAString& aName, const nsAString& aValue,
+                              bool aSession, int64_t aExpires,
+                              const nsAString& aPath, int32_t aSameSite,
+                              bool aPartitioned, const nsID& aOperationID);
 
   // Returns true if a cookie notification has been generated while completing
   // the operation.
   bool DeleteRequestOnMainThread(
       ThreadsafeContentParentHandle* aParent, const nsAString& aDomain,
       const OriginAttributes& aOriginAttributes, bool aThirdPartyContext,
-      bool aPartitionForeign, bool aUsingStorageAccess, const nsAString& aName,
+      bool aPartitionForeign, bool aUsingStorageAccess,
+      bool aIsOn3PCBExceptionList, const nsAString& aName,
       const nsAString& aPath, bool aPartitioned, const nsID& aOperationID);
 
   CookieStoreNotificationWatcher* GetOrCreateNotificationWatcherOnMainThread(